### PR TITLE
git-r3.eclass: Inroducing EGIT_DIR_NAME var to record all the repos being fetched in the gir3_src

### PR DESCRIPTION
**Pull request by siddhanthrathod**
see_also: https://github.com/gentoo/gentoolkit/pull/33

### CODE REVIEW

**Review by mgorny — CHANGES_REQUESTED** [review]
That's an awful hack with low quality code, zero effort to justify it and no documentation.

**Comment by siddhanthrathod** [comment]
> That's an awful hack with low quality code, zero effort to justify it and no documentation.

I'm sorry I'll do better

File: eclass/git-r3.eclass [hunk 1]
```diff
@@ -120,6 +120,12 @@ fi
 # usage.
 : "${EGIT_LFS_CLONE_TYPE:=shallow}"
 
+# @ECLASS_VARIABLE: GDIR_NAME
```
Comment (mgorny) [inline eclass/git-r3.eclass:123]: 1. Prefix it properly with `EGIT_`.
2. Mark as `@OUTPUT_VARIABLE`.
3. Make it an array.

**Review by mgorny — COMMENTED** [review]
That said, I'd still like to see a working code using this metadata first.

**Comment by siddhanthrathod** [comment]
https://github.com/gentoo/gentoolkit/commit/f36db1ec472539ee159c60a2aeede1fdaca69a51
the code utilizing it 
https://github.com/gentoo/gentoolkit/pull/33#issuecomment-1741852691
working prototype

File: eclass/git-r3.eclass [hunk 2]
```diff
@@ -120,6 +120,13 @@ fi
 # usage.
 : "${EGIT_LFS_CLONE_TYPE:=shallow}"
 
+# @ECLASS_VARIABLE: EGIT_DIR_NAME
```
Comment (mgorny) [inline eclass/git-r3.eclass:123]:
```suggestion
# @ECLASS_VARIABLE: EVCS_STORE_DIRS
```

File: eclass/git-r3.eclass [hunk 3]
```diff
@@ -120,6 +120,13 @@ fi
...
+# @OUTPUT_VARIABLE
+# @DESCRIPTION:
+# Record of names of all the repositories directories being cloned in the git3_src.
+# This is useful in the case of ebuild that fetch multiple repos.
```
Comment (mgorny) [inline eclass/git-r3.eclass:127]: It would be better to say that it can be used by `eclean` to clean them up.

File: eclass/git-r3.eclass [hunk 4]
```diff
@@ -355,6 +362,8 @@ _git-r3_set_gitdir() {
 	# now replace all the slashes
 	repo_name=${repo_name//\//_}
 
+	EGIT_DIR_NAME+=("${repo_name}")
```
Comment (mgorny) [inline eclass/git-r3.eclass:365]: Put full path so you wouldn't have to deal with different values of `EGIT3_STORE_DIR`, and so you wouldn't have to reinvent it for every VCS out there.

**Review by mgorny — CHANGES_REQUESTED** [review]

**Comment by siddhanthrathod** [comment]
@mgorny thanks for the suggestions I have made the changes as you requested Do they look okay ?

File: eclass/git-r3.eclass [hunk 5]
```diff
@@ -360,6 +368,8 @@ _git-r3_set_gitdir() {
 
 	GIT_DIR=${EGIT3_STORE_DIR}/${repo_name}
 
+	EVCS_STORE_DIRS+=("${GIT_DIR}")
```
Comment (mgorny) [inline eclass/git-r3.eclass:371]:
```suggestion
	EVCS_STORE_DIRS+=( "${GIT_DIR}" )
```

File: eclass/git-r3.eclass (hunk 4 above)
Comment (floppym) [reply eclass/git-r3.eclass]: The full path will break if the user moves DISTDIR after they have installed a git-r3-based package.

Comment (mgorny) [reply eclass/git-r3.eclass]: There is no requirement that `EGIT3_STORE_DIR` is under `DISTDIR`. In fact, I use a completely separate top-level directory for all VCS caches.

**Pull request closed without merging**